Core/src/Material/SurfaceMaterialMapper.cpp

```diff
@@ -199,8 +199,33 @@ void Acts::SurfaceMaterialMapper::finalizeMaps(State& mState) const {
 
 void Acts::SurfaceMaterialMapper::mapMaterialTrack(
     State& mState, RecordedMaterialTrack& mTrack) const {
-  using VectorHelpers::makeVector4;
+  // Retrieve the recorded material from the recorded material track
+  auto& rMaterial = mTrack.second.materialInteractions;
+  ACTS_VERBOSE("Retrieved " << rMaterial.size()
+                            << " recorded material steps to map.")
 
+  // Check if the material interactions are associated with a surface. If yes we
+  // simply need to loop over them and accumulate the material
+  if (rMaterial.begin()->intersectionID != GeometryIdentifier()) {
```

Comment:
`intersectionID` would only different from `GeometryIdentifier()` if the root reader was used with the `readSurface` option. This mean that surface information has been associated with the material interaction and this would be the same across the entire root file. Basically I use this as a proxy to check if the root reader was used with `readSurface=true`.